XSConsoleData.py

```diff
@@ -528,17 +545,95 @@ def SaveToNTPConf(self):
         # Double-check authentication
         Auth.Inst().AssertAuthenticated()
 
-        file = None
         try:
-            file = open("/etc/chrony.conf", "w")
-            for other in self.ntp.othercontents([]):
-                file.write(other+"\n")
-            for server in self.ntp.servers([]):
-                file.write("server "+server+" iburst\n")
+            with open("/etc/chrony.conf", "w") as confFile:
+              for other in self.ntp.othercontents([]):
+                  confFile.write(other + "\n")
+              for server in self.ntp.servers([]):
+                  confFile.write("server " + server + " iburst\n")
         finally:
-            if file is not None: file.close()
             self.UpdateFromNTPConf()
 
+        # Force chronyd to update the time
+        if self.data['ntp']['method'] != "Disabled":
+            servers = self.data['ntp']['servers']
+            if self.data['ntp']['method'] == "DHCP":
+                servers = self.GetDHClientInterfaces()
+
+            if servers:
+                self.StopService("chronyd")
+                getoutput("chronyd -q 'server %s iburst'" % servers[0])  # Update the clock and exit
+                getoutput("hwclock -w")  # Sync hwclock with date set by chronyd
+                self.StartService("chronyd")
```

Comment:
I'm not familiar with `chronyd`.
I think, these comments would help people like me to understand what is done here:
```suggestion
            if servers:
                self.StopService("chronyd")
                # Single shot time update like: `ntpdate servers[0]`
                getoutput("chronyd -q 'server %s iburst'" % servers[0])
                # Write the system time set by the single shot NTP to the HW clock
                getoutput("hwclock -w")  
                self.StartService("chronyd")
```